### ISSUE TITLE
[no-call-expression] triggers on signal inputs

### ISSUE THREAD

**Issue by Merriz** [angular-eslint/angular-eslint#2911]
Should the rule really trigger on a SignalInput like this?:

TS:
`public id = input<string>();`

HTML:
`<update-offer-component [offerId]="id()" />`

We added this rule to our repo in hopes that it would be a good safeguard against calling expressions too many times on every change detection cycle, but instead we're faced with errors everywhere related to signal-based inputs in our custom components.

My understanding is that this isn't intentional and just an oversight in the detection.

**Versions**

| package                                  | version |
| ---------------------------------------- | ------- |
| `@angular-eslint/eslint-plugin-template` | `20.3.0` |
| `@angular-eslint/template-parser`        | `20.3.0` |
| `@typescript-eslint/parser`              | `8.40.0` |
| `ESLint`                                 | `9.39.1` |
| `node`                                   | `22.15.0` |

```sh
Package                            Version
------------------------------------------
@angular-devkit/build-angular      20.3.11
@angular-devkit/core               20.3.11
@angular-devkit/schematics         20.3.11
@angular/cdk                       20.2.14
@angular/cli                       20.3.12
@angular/compiler                  20.3.16
@angular/material                  20.2.14
@angular/material-moment-adapter   20.2.14
@schematics/angular                20.3.11
rxjs                               7.8.2
typescript                         5.9.3
zone.js                            0.15.1
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by json-derulo** [comment]
This is expected, due to limitations of the template parsing. For a detailed explanation, see https://github.com/angular-eslint/angular-eslint/issues/1380#issuecomment-1783783808

**Comment by Merriz** [comment]
Thanks for the explanation. We'll make do without this rule then! Closing the issue.
